### PR TITLE
[441] Add progress/state for degree section

### PR DESCRIPTION
**Pull request by jhenderson**
### Context
https://trello.com/c/O9gm5lcw/441-degree-details-progress-state

### Changes proposed in this pull request
Add progress/state tracking for the degree section

### Demo
483_degree_progress_state

### CODE REVIEW

File: app/models/degree_detail.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class DegreeDetail
+  include ActiveModel::Model
+
+  attr_accessor :trainee, :degree_ids
+
+  validate :degrees_cannot_be_empty
+
+  def initialize(trainee)
+    @trainee = trainee
+    super(degree_ids: trainee.degree_ids)
```
Comment (darokel) [inline app/models/degree_detail.rb:10]:
```suggestion
super(fields)
```

**Review by darokel — APPROVED** [review]
I left a suggestion but looks good otherwise

File: app/models/degree_detail.rb (hunk 1 above)
Comment (jhenderson) [reply app/models/degree_detail.rb]: Nice spot 👍

File: spec/features/trainees/creating_a_new_trainee_spec.rb [hunk 2]
```diff
@@ -15,13 +15,12 @@
 private
 
   def when_i_am_viewing_the_list_of_trainees
-    @index_page ||= PageObjects::Trainees::Index.new
-    @index_page.load
+    trainee_index_page.load
```
Comment (aliuk2012) [inline spec/features/trainees/creating_a_new_trainee_spec.rb:18]: Why are we replacing SitePrism pages?

**Comment by defong** [comment]
[Degree details progress/state](https://trello.com/c/O9gm5lcw/441-degree-details-progress-state)

File: app/models/degree_detail.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  validate :degrees_cannot_be_empty
+
+  def initialize(trainee)
+    @trainee = trainee
+    super(fields)
+  end
+
+  def fields
+    { degree_ids: trainee.degree_ids }
+  end
```
Comment (defong) [inline app/models/degree_detail.rb:15]: Should be a private method right?

Comment (jhenderson) [reply app/models/degree_detail.rb:15]: Negative. It's called here `app/services/progress_service.rb:21`

File: spec/features/trainees/creating_a_new_trainee_spec.rb (hunk 2 above)
Comment (jhenderson) [reply spec/features/trainees/creating_a_new_trainee_spec.rb:18]: We're not. There's a lot of duplicate code in the feature specs. We can DRY it up a bit using some modules. Example: `spec/support/features/trainee_steps.rb`.

**Review by gpeng — APPROVED** [review]

**Pull request merged**